controllers/metal3.io/host_state_machine.go

```diff
@@ -240,13 +240,15 @@ func (hsm *hostStateMachine) handleRegistering(info *reconcileInfo) actionResult
 	} else {
 		hsm.NextState = metal3v1alpha1.StateInspecting
 	}
+	hsm.Host.Status.ErrorCount = 0
```

Comment:
Probably it could make sense to bind the error count reset to the `actionComplete` (maybe with an utility method like `recordActionComplete`), otherwise my feeling is that it could be easily missed next time some new code will be added handling a completed action.

Reply:
It was nice having it be magic, but there's no current place to hang it - actionComplete doesn't work for the steady states.
The fact that we can't move the code into the state machine for the steady states indicates a design problem, and maybe once that is resolved we'll have a convenient place to tie it.

Reply:
Agree that there's probably something to be reviewed in the steady states, as in the current implementation the ErrorCount is cleared when:
1. When completing an action and moving from/to: 
  * Registering -> Inspecting/Ext.Prov
  * Inspecting -> Match Profile
  * Provisioning -> Provisioned
  * Deprovisioning -> Deleting / Ready
2. Due the power management within the steady states (Ext.Prov.//Provisioned ) and Ready

But at least for point 1 having an utility function that set the actionComplete and resets the ErrorCount could help in reducing the scattering.